src/main/java/io/cdap/plugin/WindowAggregationConfig.java

```diff
@@ -0,0 +1,406 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import org.apache.parquet.Strings;
+import org.apache.spark.sql.Column;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Config for Window aggregation plugin.
+ */
+public class WindowAggregationConfig extends PluginConfig {
+
+  @Macro
+  @Description("Specifies a list of fields, comma-separated, to partition the data by. At least 1 field must be " +
+    "provided.")
+  private String partitionFields;
+
+  @Macro
+  @Nullable
+  @Description("Specifies key-value pairs containing the ordering field, and the order "
+    + "(ascending or descending). Sorts numerically for numeric order by field, lexicographically for strings.")
+  private String partitionOrder;
+
+  @Macro
+  @Nullable
+  @Description("Selects the type of window frame to create within each partition. "
+    + "Options can be ROW or RANGE or NONE.")
+  private String windowFrameType;
+
+  @Macro
+  @Nullable
+  @Description("Whether to use an unbounded start boundary for a frame.")
+  private String unboundedPreceding;
+
+  @Macro
+  @Nullable
+  @Description("Whether to use an unbounded end boundary for a frame.")
+  private String unboundedFollowing;
+
+  @Macro
+  @Nullable
+  @Description("Specifies the number of preceding rows in the window frame. When Frame Type is ROW, this is a number" +
+    " relative to the current row. E.g. -2 to begin the frame 2 rows before the current row. When Frame Type is RANGE,"
+    + "specifies the value to be subtracted from the value of the current row to get the start boundary.")
+  private String preceding;
+
+  @Macro
+  @Nullable
+  @Description("Specifies the number of following rows to include in the window frame. When Frame Type is ROW, this " +
+    "is a number relative to the current row. E.g. 3 to end the frame 3 rows after the current row. When Frame Type " +
+    "is RANGE, specifies the value to be added to the value of the current row to get the end boundary.")
```

Comment:
How is ROW different than range?

Reply:
**Row:** A row based boundary is based on the position of the row within the partition. An offset indicates the number of rows above or below the current row, the frame for the current row starts or ends. For instance, given a row based sliding frame with a lower bound offset of -1 and a upper bound offset of +2. The frame for row with index 5 would range from index 4 to index 6.



**Range:** A range-based boundary is based on the actual value of the ORDER BY expression(s). An offset is used to alter the value of the ORDER BY expression, for instance if the current order by expression has a value of 10 and the lower bound offset is -3, the resulting lower bound for the current row will be 10 - 3 = 7. This however puts a number of constraints on the ORDER BY expressions: there can be only one expression and this expression must have a numerical data type. An exception can be made when the offset is unbounded, because no value modification is needed, in this case multiple and non-numeric ORDER BY expression are allowed.

 Reference:
 https://spark.apache.org/docs/2.3.0/api/java/org/apache/spark/sql/expressions/WindowSpec.html#rowsBetween-long-long-
 https://spark.apache.org/docs/2.3.0/api/java/org/apache/spark/sql/expressions/WindowSpec.html#rangeBetween-long-long-